packages/particle-network/src/index.ts

```diff
@@ -0,0 +1,127 @@
+import type {
+  WalletInit,
+  EIP1193Provider,
+  ProviderRpcError,
+  ProviderRpcErrorCode,
+  ProviderAccounts
+} from '@web3-onboard/common'
+import type { Config } from '@particle-network/auth'
+
+/**
+ * Represents the different authentication methods available.
+ * @typedef {'email' | 'phone' | 'google' | 'apple' | 'twitter' | 'facebook' | 'microsoft' | 'linkedin' | 'github' | 'twitch' | 'discord'} AuthTypes
+ */
+type AuthTypes =
+  | 'email'
+  | 'phone'
+  | 'google'
+  | 'apple'
+  | 'twitter'
+  | 'facebook'
+  | 'microsoft'
+  | 'linkedin'
+  | 'github'
+  | 'twitch'
+  | 'discord'
+
+/**
+ * Interface for setting a preferred social login type and whether it should be displayed or just routed through the standard menu item.
+ * @interface
+ */
+interface PreferredAuthType {
+  type: AuthTypes
+  setAsDisplay: boolean
+}
+
+/**
+ * Configuration options enabling custom authentication type selection, extending the basic Config.
+ * @interface
+ * @extends {Config}
+ */
+interface ParticleAuthModuleOptions extends Config {
+  preferredAuthType?: AuthTypes | PreferredAuthType
+}
+
+const getDisplayLabel = (authType?: string, shouldSetDisplay?: boolean) => {
+  if (authType) {
+    return shouldSetDisplay
+      ? authType.charAt(0).toUpperCase() + authType.slice(1)
+      : 'Particle Network'
+  }
+  return 'Particle Network'
+}
+
+const particleAuth = async (
+  options: ParticleAuthModuleOptions
+): Promise<WalletInit> => {
+  const { createEIP1193Provider } = await import('@web3-onboard/common')
+  const { ParticleNetwork } = await import('@particle-network/auth')
+  const { ParticleProvider } = await import('@particle-network/provider')
+
```

Comment:
Can we move these into the `getInterface` function so they are only imported once a use selects particle?

Reply:
Sure! Just shifted it